sample/src/main/java/co/nimblehq/sample/ui/surveylist/SurveyListActivity.kt

```diff
@@ -0,0 +1,36 @@
+package co.nimblehq.sample.ui.surveylist
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import co.nimblehq.sample.databinding.ActivitySurveyListBinding
+
+class SurveyListActivity : AppCompatActivity() {
+
+    private lateinit var surveyListViewModel: SurveyListViewModel
+    private lateinit var binding: ActivitySurveyListBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = ActivitySurveyListBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        title = "Survey List"
+
+        surveyListViewModel = ViewModelProvider(this, SurveyListModelFactory())
+            .get(SurveyListViewModel::class.java)
+
+        surveyListViewModel.surveyListResult.observe(this@SurveyListActivity, {
+            it.success?.let { result ->
+                (binding.surveyListView.adapter as SurveyAdapter).updateData(result.listSurvey)
+            }
+
+        })
+        binding.surveyListView.adapter = SurveyAdapter()
+
+        surveyListViewModel.getSurveyList()
```

Comment:
I think you can put it at the `init()` of the ViewModel 🆗

Reply:
@sleepylee I don't think so. It is better for us to control the flow, to make sure that all stuff things are ready before fetching data.

Reply:
@doannimble 🤔  a ViewModel can outlive the View, let's say you're rotating the screen and recreate the Activity, it will jump to this methods again, and the result will be the same but it costs you 1 extra fetch for the same list of survey 🤷 
Putting this in ViewModel will fetch your data ready, and it doesn't publish the data directly to the View, so you don't have to worry about: `make sure that all stuff things are ready` - you don't have to care, the data is available at `LiveData` - the faster you load the better, when View wants to get the data, just `observe`. When View is destroyed/re-create? ViewModel data won't have to care either, they keep the data persisted 😄  let me know if you get the points or finding anything opposite

Reply:
@sleepylee I've just made a quick review from ViewModel, and see the way that Google suggest us to use :
```
private val users: MutableLiveData<List<User>> by lazy {
        MutableLiveData<List<User>>().also {
            loadUsers()
        }
    }
```
And the idea of this look like yours. So we will go ahead with the way Google suggest instead of using `init`?

Reply:
could you please paste link to the `suggestion` from google here for ref? I do believe it's not a single way to do, while it's valid: lazy block will init once only when there is a subscription started to observe, I do think that my solution was also valid to cases where you want to eagerly loading the data before subscription happening. I also took it from some architecture-sample repos from Google back to the day. 

Like I said, it would work if you apply the above suggestion, I'm not against it. My point here is to pay attention at the action triggered from the View compared to your original implementation. If it was a click event that init the flow, then it's ok for the View to control the behavior trigger, if it was not triggered by an action, then it shouldn't be controlled by the View's lifecycle

Reply:
@sleepylee for your reference : https://developer.android.com/topic/libraries/architecture/viewmodel
I understand the different. Just wondering which one is better?

Reply:
here is my reference, @doannimble https://github.com/android/architecture-samples/blob/dev-hilt/app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksViewModel.kt#L100

Reply:
@sleepylee ok, as there is no action to trigger the request. Then put it inside the `init` of ViewModel

Reply:
@doannimble as far as I understand, there is no solution better than the other, they both aimed to this target:

> make sure to assign responsibility to acquire and keep the list of users to a ViewModel

The difference I can see is between lazy & eager load.
- Field lazy { } initialization will happen only when there is a reference to it. In the context of a ViewModel where it has many Views subscribe to it, not all of them require all the fields to be initialized at the same time. 
For example: 

- Given a `UserViewModel` that manage both `token credential` and `user profile`, and we have 2 different Views bound to this ViewModel: `LoginStateFragment` and `UserProfileFragment` . The ViewModel is retained, the user might entering the LoginStateFragment but maybe they don't go further to the UserProfileFragment; in that case, `val userProfile = lazy { }` will not be invoked, so no API is called when it's not needed.
- Now if we talk about a different case where the initialization of a field, let's say `credential` in that `UserViewModel`is needed everywhere, and 100% there is a usage every time, then we can let it initialize the request first at `init` (ViewModel's init is called before View's onCreate if you start the ViewModel binding early). In case there could be some animation, UI changing before the actual data rendering, you can make sure that you data will be ready right after the animation is completed.

Reply:
@sleepylee Thanks for your explanation. Now I know more purpose of calling actions from `init` of ViewModel.